File: stalin_sort.nim
import std/random

proc stalin_sort(arr: seq[float64]): seq[float64] =
    if arr.len == 0:
        return arr
    var
        sorted = newSeq[float64]()
        max_v = arr[0]
    for i in arr:
        if i >= max_v:
            sorted.add(i)
            max_v = i
        else:
            echo i, " sent to Gulag"
    return sorted

randomize()
var
    r = initRand()
    sq1 = newSeq[float64]()
    #re = r.rand(100.0)
for j in 0..50:
    let re = r.rand(100.0)
    sq1.add(re)
echo sq1
echo stalin_sort(sq1)

File: stalin_sort.jl
function stalin_sort(inp)
    if length(inp) == 0
        return inp
    end
    max_v = inp[1]
    sorted = []

    for i in inp
        if i >= max_v
            push!(sorted, i)
            max_v = i
        else
            println("$i sent to Gulag")
        end
    end

    return sorted
end

A = []

for j =  1:50
    R = 100 * rand(Float64)
    push!(A, R)
end

println(A)
println(stalin_sort(A))
